sockeye/train.py

```diff
@@ -767,35 +820,36 @@ def train(args: argparse.Namespace) -> training.TrainState:
             min_epochs=args.min_num_epochs,
             max_epochs=args.max_num_epochs,
             update_interval=args.update_interval,
-            stop_training_on_decoder_failure=args.stop_training_on_decoder_failure
+            stop_training_on_decoder_failure=args.stop_training_on_decoder_failure,
+            using_horovod=using_horovod
         )
         if trainer_config.min_epochs is not None and trainer_config.max_epochs is not None:
             check_condition(trainer_config.min_epochs <= trainer_config.max_epochs,
                             "Minimum number of epochs must be smaller than maximum number of epochs")
 
-        # Fixed training schedule always runs for a set number of updates
-        if args.learning_rate_schedule:
-            trainer_config.min_updates = None
-            trainer_config.max_updates = sum(num_updates for (_, num_updates) in args.learning_rate_schedule)
-            trainer_config.max_num_checkpoint_not_improved = -1
-            trainer_config.min_samples = None
-            trainer_config.max_samples = None
-            trainer_config.min_epochs = None
-            trainer_config.max_epochs = None
-
         optimizer_config = create_optimizer_config(args)
         training_model.initialize(optimizer_config.initializer, ctx=context)
         if args.params is not None:  # load existing parameters if present
             training_model.load_parameters(filename=args.params,
                                            ctx=context,
-                                           allow_missing=args.allow_missing_params or model_config.lhuc)
+                                           allow_missing=args.allow_missing_params or model_config.lhuc,
+                                           cast_dtype=True)
```

Comment:
could you also make explicit the other argument (`dtype_source`) here?

Reply:
Are you referring to passing `dtype='current'`?

Reply:
yes `dtype_source='current'` instead of `saved', meaning that it will cast the on-disk parameter to the precision of the Model block and not vice versa.